### PR TITLE
[#818] better decoupling from standalone and eclipse astparserfactory

### PR DESCRIPTION
**Pull request by cdietrich**
[#818] better decoupling from standalone and eclipse astparserfactory
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtend.ide/src/org/eclipse/xtend/ide/javaconverter/EclipseASTParserFactory.xtend [hunk 1]
```diff
@@ -34,7 +34,7 @@ class EclipseASTParserFactory extends org.eclipse.xtend.core.javaconverter.ASTPa
 			}
 			return new ASTParserWrapper(projlevel, parser)
 		}
-		return super.createJavaParser(context)
+		throw new IllegalStateException("did not find java project")
```
Comment (cdietrich) [inline org.eclipse.xtend.ide/src/org/eclipse/xtend/ide/javaconverter/EclipseASTParserFactory.xtend]: i wonder which usecase this addressed and what we should do here.
if there is no java project does it make sense to use the classpath of xtend.ide?
(beside adding clasgraph to orbit)

**Comment by cdietrich** [comment]
Looks like that does not completely help
https://ci.eclipse.org/xtext/job/xtext-xtend/job/cd_issue818/2/console

**Comment by cdietrich** [comment]
looks like there are ide tests that use runtime injector
org.eclipse.xtend.ide.tests.AbstractXtendTestCase

**Comment by cdietrich** [comment]
=> Options are:
- find solution
- revert changes and find clean solution
- add classgraph to orbit (might be needed for @ArneDeutsch anyway)

**Comment by szarnekow** [comment]
I’ll file the CQ for orbit

**Comment by cdietrich** [comment]
Obsolete

**Pull request closed without merging**